crates/biome_js_semantic/src/events.rs

```diff
@@ -365,150 +364,159 @@ impl SemanticEventExtractor {
+                        }
+                        self.push_binding(None, BindingName::Type(name), info);
+                    }
+                    AnyJsBindingDeclaration::JsArrowFunctionExpression(_)
+                    | AnyJsBindingDeclaration::JsBogusParameter(_)
+                    | AnyJsBindingDeclaration::JsFormalParameter(_)
+                    | AnyJsBindingDeclaration::JsRestParameter(_)
+                    | AnyJsBindingDeclaration::TsIndexSignatureParameter(_)
+                    | AnyJsBindingDeclaration::TsPropertyParameter(_)
+                    | AnyJsBindingDeclaration::JsCatchDeclaration(_) => {
+                        self.push_binding(None, BindingName::Value(name), info);
+                    }
+                    AnyJsBindingDeclaration::TsInferType(_) => {
+                        // Delay the declaration of parameter types that are inferred.
+                        // Their scope corresponds to the true branch of the conditional type.
+                        self.infers
+                            .push(TsTypeParameterName::unwrap_cast(node.syntax().clone()));
+                        return;
                     }
-                    self.push_binding(None, BindingName::Type(name), info);
-                }
-                AnyJsBindingDeclaration::JsArrowFunctionExpression(_)
-                | AnyJsBindingDeclaration::JsBogusParameter(_)
-                | AnyJsBindingDeclaration::JsFormalParameter(_)
-                | AnyJsBindingDeclaration::JsRestParameter(_)
-                | AnyJsBindingDeclaration::TsIndexSignatureParameter(_)
-                | AnyJsBindingDeclaration::TsPropertyParameter(_)
-                | AnyJsBindingDeclaration::JsCatchDeclaration(_) => {
-                    self.push_binding(None, BindingName::Value(name), info);
-                }
-                AnyJsBindingDeclaration::TsInferType(_) => {
-                    // Delay the declaration of parameter types that are inferred.
-                    // Their scope corresponds to the true branch of the conditional type.
-                    self.infers
-                        .push(TsTypeParameterName::unwrap_cast(node.syntax().clone()));
-                    return Some(());
                 }
+                is_exported
+            } else {
+                // Handle identifiers in bogus nodes,
+                self.push_binding(None, BindingName::Value(name), info);
+                false
             }
-            is_exported
         } else {
-            // Handle identifiers in bogus nodes,
-            self.push_binding(None, BindingName::Value(name), info);
+            // The binding has a bogus name
             false
         };
         let scope_id = self.current_scope_mut().scope_id;
         self.stash.push_back(SemanticEvent::DeclarationFound {
             scope_id,
             hoisted_scope_id,
-            name_token,
+            range: node.syntax().text_range(),
```

Comment:
Is this intentional? Should it be `text_trimmed_range`?

Reply:
It is intentional. We use `text_range` everywhere.
I still plan to change it to `text_trimmed_range`.